src/tools/rustfmt/src/items.rs

```diff
@@ -1182,7 +1183,8 @@ pub(crate) fn format_trait(
 
     let mut result = String::with_capacity(128);
     let header = format!(
-        "{}{}{}trait ",
+        "{}{}{}{}trait ",
+        format_constness(constness),
```

Comment:
https://github.com/rust-lang/rustfmt/issues/6619 was recently opened. @fee1-dead I think the formatting with regards to visibility is incorrect. I tested this with rustfmt 1.8.0-nightly (3048886e59 2025-07-30)